npm/webpack-dev-server/src/loader.ts

```diff
@@ -18,7 +18,7 @@ const makeImport = (file: Cypress.Cypress['spec'], filename: string, chunkName:
   const magicComments = chunkName ? `/* webpackChunkName: "${chunkName}" */` : ''
 
   return `"${filename}": {
-    shouldLoad: () => decodeURI(document.location.pathname).includes("${file.absolute}"),
+    shouldLoad: () => (new URL(document.location)).searchParams.get("specPath") === "${file.absolute}",
```

Comment:
I think this should be fine, let's see how CI goes.

Reply:
It works, however, this makes it a breaking change I think. This will become a problem if the cypress-version doesn't match the webpack-dev-server version, and I don't know if I can assume this to always be the case.

Reply:
Why is this a breaking change? Can you share some examples of when `shouldLoad` will be return a different value after your change?

Making a breaking change is definitely not something we take lightly, I think we want to avoid this if possible. Otherwise, this change would need to wait for the next major - a lot of planning goes into these.

Reply:
This is only a breaking change if a url that previously returned `true` for `shouldLoad` is now `false`, or vice versa. Is there an example of this?

As long as `shouldLoad` does the same thing, regardless of how it is implemented, it shouldn't be a breaking change.

Reply:
I'd missed that `document.pathname` does NOT include search params. I understand the breaking change now.

Reply:
I extended the table, should I also update the changelog?

Reply:
The changelog for npm is automated, no problem here.